src/feature/dircache/dircache_stub.c

```diff
@@ -0,0 +1,152 @@
+/* Copyright (c) 2001-2004, Roger Dingledine.
+ * Copyright (c) 2004-2006, Roger Dingledine, Nick Mathewson.
+ * Copyright (c) 2007-2020, The Tor Project, Inc. */
+/* See LICENSE for licensing information */
+
+/**
+ * @file dircache_stub.c
+ * @brief Stub declarations for use when dircache module is disabled.
+ **/
+
+#include "core/or/or.h"
+#include "feature/dircache/consdiffmgr.h"
+#include "feature/dircache/dircache.h"
+#include "feature/dircache/dirserv.h"
+#include "feature/dircommon/dir_connection_st.h"
+
+int
+directory_handle_command(dir_connection_t *conn)
+{
+  (void) conn;
+  tor_assert_nonfatal_unreached_once();
+  return -1;
+}
+
+int
+connection_dirserv_flushed_some(dir_connection_t *conn)
+{
+  (void) conn;
+  tor_assert_nonfatal_unreached_once();
+  return -1;
+}
+
+int
```

Comment:
We usually stub out functions like this using macros, because it lets the optimiser eliminate more dead code. This comment applies to all the stub functions that return a value, particularly if they have no side effects.

(The functions with side effects could also become inline functions in the header.)

For example:
https://github.com/torproject/tor/blob/master/src/feature/relay/routermode.h#L28